src/Sender/CreditMemoEmailSender.php

```diff
@@ -22,36 +22,34 @@ final class CreditMemoEmailSender implements CreditMemoEmailSenderInterface
 {
     private const UNITS_REFUNDED = 'units_refunded';
 
-    private CreditMemoPdfFileGeneratorInterface $creditMemoPdfFileGenerator;
-
-    private SenderInterface $sender;
-
-    private FileManagerInterface $fileManager;
-
     public function __construct(
-        CreditMemoPdfFileGeneratorInterface $creditMemoPdfFileGenerator,
-        SenderInterface $sender,
-        FileManagerInterface $fileManager
+        private CreditMemoPdfFileGeneratorInterface $creditMemoPdfFileGenerator,
+        private SenderInterface $sender,
+        private FileManagerInterface $fileManager,
+        private bool $hasEnabledPdfFileGenerator = true
     ) {
-        $this->creditMemoPdfFileGenerator = $creditMemoPdfFileGenerator;
-        $this->sender = $sender;
-        $this->fileManager = $fileManager;
     }
 
     public function send(CreditMemoInterface $creditMemo, string $recipient): void
     {
-        $creditMemoPdfFile = $this->creditMemoPdfFileGenerator->generate($creditMemo->getId());
+        if ($this->hasEnabledPdfFileGenerator) {
```

Comment:
You can reverse check here and just send an email without attachment when the flag is disabled and do everything else when it is enabled